### DISCUSSION TITLE
Select Default Heat Source from Home Assistant?

### DISCUSSION THREAD

**Discussion by barryperfect** [tomasmcguinness/homeassistant-mixergy#15] [Q&A] [answered]
We will be getting a Mixergy Tank installed in 2 weeks time and we are hoping to integrate it into Home Assistant.

Does the Mixergy API allow you to change the default heat source?

Thinking of calculating the cheapest heat source (indirect or electric) and then selecting it on a daily basis.

Is this possible?

**Comment by tomasmcguinness** [comment]
Hi,

I’m pretty sure you can change the default heat source. It’s not supported by my extension, but I believe I could add that!

Let me look!

**Reply by barryperfect** [reply]
That would be great Tom! 

Happy to help test it out when our tank is installed 😀

**Reply by barryperfect** [reply]
Now have the Mixergy Tank installed 👍 Thanks for the integration, just let me know if you want any Beta versions testing 😀

**Reply by barryperfect** [reply]
@tomasmcguinness Did you ever get a chance to look at setting the default heat source via Home Assistant? 

Looking at comparing Gas and Electric heating costs it is pretty close between the two and it would be good to pick them automatically using Home Assistant.

**Comment by tomasmcguinness** [comment]
Sorry, it completely slipped my mind!

I just had a quick look at it seems like I can set the default heat source. I'll have a shot at making those changes over the weekend.

<img width="238" alt="image" src="https://github.com/tomasmcguinness/homeassistant-mixergy/assets/302741/32fbb70c-88b4-4a69-85e5-7a9cd4a892e9">

**Reply by barryperfect** [reply]
Thanks Tom,

Looks promising, assume there is a matching PUT to go with the GET?

**Reply by barryperfect** [reply]
Hi Tom,

Did you ever get anywhere with setting the default heat source? 

I see from their API it is possible, although the setting appears to be embedded with setting heating schedules, rather than as a standalone call which makes things a bit more complicated, guess you have to read the existing schedule settings and then just change the default source when setting it again?

**Reply by tomasmcguinness** [reply]
I ran into some weird behaviour and it slipped my mind!

I'll have a look again as soon as I can.

**Reply by barryperfect** [reply]
OK, thanks Tom, just checking it wasn't in the code and I didn't know :-)

Happy to test when you get the opportunity

**Reply by michaelmarconi** [reply]
Hi @tomasmcguinness — I'm interested in this too, so I'll keep an eye on this.  In my case, I'd like to be able to do a partial tank top-up using electricity, not the heat pump, which really only works if you heat up the entire tank...

**Reply by barryperfect** [reply]
@tomasmcguinness I don't suppose you ever had a chance to look at this?

At the moment it makes sense to heat the tank off peak with Electricity, but if I need to heat during the day it is better to heat with Gas (cost wise). At the moment if I want to do this I have to remember to do this manually change the source via the app.

**Comment by DirkFace** [comment]
This would be handy to have, now that octopus are having the power hour ... would be able to switch from gas to leccy & top up the tank for free all within home assistant. Saves messing about in the mixergy app & switching it over & then back again at the end of the power hour.

**Comment by tomasmcguinness** [comment, marked as answer]
It's taken me long enough ... 

![image](https://github.com/user-attachments/assets/fd241a6f-d75a-4acd-b13d-7347f997d93a)

I just to remember how to release this to HACS, but it's up in Github now!!!

**Reply by tomasmcguinness** [reply]
Just in time for today's free Octopus hour :)

**Reply by DirkFace** [reply]
Thank you tom, I'll add it to my power hour automation. Lovely jubbly

**Reply by tomasmcguinness** [reply]
Please, give it a try and report any issues!

**Reply by DirkFace** [reply]
Worked perfectly. Switched from gas to leccy, set a charge & will switch back to gas source after power hour. Don't have to change anything at the start of power hour now - thanks ... car is charging, hot water topping up, solar battery, infra red heating, washing machine, & cooker all making use of the free leccy.

**Reply by barryperfect** [reply]
Amazing @tomasmcguinness , has been working for me, I have lots of uses for this, thank you so much!

**Reply by tomasmcguinness** [reply]
https://ko-fi.com/tomasmcguinness 😜

**Reply by barryperfect** [reply]
> https://ko-fi.com/tomasmcguinness 😜

Would love to support, there doesn’t seem to be a donate button when I follow that link, I even created a ko-fi account 🙄

**Reply by tomasmcguinness** [reply]
Thanks! You can try now? My account was messed up!

**Reply by barryperfect** [reply]
Done 😀👍
